Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/CoronaTestRepo.kt

```diff
@@ -0,0 +1,50 @@
+package de.rki.coronawarnapp.coronatest
+
+import de.rki.coronawarnapp.coronatest.qrcode.CoronaTestGUID
+import de.rki.coronawarnapp.coronatest.qrcode.CoronaTestQRCode
+import de.rki.coronawarnapp.coronatest.server.CoronaTestServer
+import de.rki.coronawarnapp.coronatest.storage.CoronaTestStorage
+import de.rki.coronawarnapp.coronatest.type.CoronaTest
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.emptyFlow
+import timber.log.Timber
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class CoronaTestRepo @Inject constructor(
+    private val storage: CoronaTestStorage,
+    private val server: CoronaTestServer,
+) {
+
+    val coronaTests: Flow<Set<CoronaTest>> = emptyFlow()
+
+    /**
+     * When this returns and there was no exception, the test was registered and a valid registrationToken obtained.
+     * Your new test should be available via **coronaTests**.
+     */
+    suspend fun registerTest(request: CoronaTestQRCode) {
```

Comment:
should it return `registrationToken` ?

Reply:
I don't see an immediate need, we are more flexible if all data retrieval the UI needs to do, is done via the `coronaTests` flow.
This will make it easier for us to make test registration async in the future (without internet).